### DISCUSSION TITLE
Can we use typescript instead of esbuild?

### DISCUSSION THREAD

**Discussion by capaj** [egoist/tsup#620] [Q&A]
I know in general esbuild is faster, but I am using TS features which are not supported by esbuild(decorators, metadata etc) so I would like to use tsup as it provides a nice CLI for building my project, but use TS as compiler. Is this possible?

**Comment by egoist** [comment]
> but use TS as compiler. Is this possible

why not use rollup with its typescript plugin then? tsup will use swc to handle `decorators/metadata`, does that work for you?

**Reply by OmgImAlexis** [reply]
This is what I've been using on all of my projects that need decorators, etc. and it works perfectly. Highly recommend @capaj
